File: Linked List/Add Two Numbers/Add_Two_Numbers.cpp
//Problem Name: Add Two Number
//Problem Link: https://leetcode.com/problems/add-two-numbers/description/

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* temp1 = l1;
        ListNode* temp2 = l2;
        ListNode* dummyNode = new ListNode(-1);
        ListNode* temp = ans;
        int sum = 0, carry = 0;
        while(temp1!=NULL || temp2!=NULL) {
            int digit1 = (temp1!=NULL) ? temp1->val:0;
            int digit2 = (temp2!=NULL) ? temp2->val:0;
            sum = digit1 + digit2 + carry;
            carry = sum/10;
            temp->next = new ListNode(sum%10);
            temp = temp->next; 
            if(temp1!=NULL) temp1 = temp1->next;
            if(temp2!=NULL) temp2 = temp2->next;
        }
        if(carry>0) {
            temp->next = new ListNode(carry);
        }
        return ans->next;
    }
};

File: Linked List/Add Two Numbers/Add_Two_Numbers.java
//Problem Name: Add Two Number
//Problem Link: https://leetcode.com/problems/add-two-numbers/description/

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode temp1 = l1;
        ListNode temp2 = l2;
        ListNode dummyNode = new ListNode(-1);
        ListNode temp = dummyNode;
        int sum = 0, carry = 0;
        while(temp1!=null || temp2!=null) {
            int digit1 = (temp1!=null) ? temp1.val : 0;
            int digit2 = (temp2!=null) ? temp2.val : 0;
            sum = digit1 + digit2 + carry;
            temp.next = new ListNode(sum%10);
            carry = sum/10;
            if(temp1!=null) temp1 = temp1.next;
            if(temp2!=null) temp2 = temp2.next;
            temp = temp.next;
        }
        if(carry>0) {
            temp.next = new ListNode(carry);
        }
        return dummyNode.next;
    }
}